packages/flutter/lib/src/material/button_style.dart

```diff
@@ -435,3 +435,19 @@ class _LerpShapes implements MaterialStateProperty<OutlinedBorder?> {
     return ShapeBorder.lerp(resolvedA, resolvedB, t) as OutlinedBorder?;
   }
 }
+
+/// How the icon in any `button.icon` should be placed, where `button` could be:
+/// * [TextButton]
+/// * [ElevatedButton]
+/// * [OutlinedButton]
+///
+/// See also:
+///
+/// * [TextButton], [ElevatedButton], [OutlinedButton], the button widgets
+enum ButtonIconAlignment {
+  /// Places the icon to the start of the button
+  start,
+
+  /// Places the icon to the end of the button
+  end,
+}
```

Comment:
the documentation here for "start" and "end" doesn't say anything you couldn't guess from the name. This violates our style guide. :-) https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-useless-documentation

Try to think of why someone would be looking at this documentation, and answer questions they may have. For example, what does "end of the button" mean? Is there padding applied? Is it inside the button or outside the button?

Reply:
I guess the new docs I just added are good, let me know what you think.